### PR TITLE
Check header for consistent instrument name

### PR DESCRIPTION
**Pull request by profxj**
For the chosen spectrograph, check that the instrument
name in the header matches expectations.  If possible.

Throw a warning if not.

Address #1276

### CODE REVIEW

**Comment by profxj** [comment]
I was having it throw an error instead of a warning,
but already found some edge cases and don't wish
the code to crash on this check.

File: pypeit/spectrographs/spectrograph.py
Comment (kbwestfall) [inline pypeit/spectrographs/spectrograph.py]: Formatting like this is going to be ignored when sphinx makes the docs.  I.e., the first and second line are just separated by a space and formatted in html.  You should treat docstrings like this as unformatted text.

File: pypeit/spectrographs/spectrograph.py
Comment (kbwestfall) [inline pypeit/spectrographs/spectrograph.py]: Here and above, I think it would be better if you didn't call `warn` twice.  Just use a long string:
```
msgs.warn("The instrument read from the headers of your data do not match the selected "
          f"spectrograph.  Found '{instr_names[0]}', expected '{self.header_name}'.  "
          "Either you're using the wrong data or the wrong spectrograph name.")
```
Note you can mix and match string types; "f-strings" can be combined with normal strings.

**Review by kbwestfall — COMMENTED** [review]
Looks great!  Just a couple of minor comments.  We should add the header names for as many of the "supported" spectrographs as we can in this PR.

File: pypeit/spectrographs/keck_lris.py
Comment (kbwestfall) [inline pypeit/spectrographs/keck_lris.py]: remove?

**Review by jhennawi — CHANGES_REQUESTED** [review]
Looks good. Is there a reason why only a handful of instruments have the header_name defined?

**Comment by profxj** [comment]
Will add some more but it is laborious to manually find these..

File: pypeit/spectrographs/spectrograph.py
Comment (profxj) [reply pypeit/spectrographs/spectrograph.py]: roger

File: pypeit/spectrographs/spectrograph.py
Comment (profxj) [reply pypeit/spectrographs/spectrograph.py]: done!

File: pypeit/spectrographs/keck_lris.py
Comment (profxj) [reply pypeit/spectrographs/keck_lris.py]: removed

**Comment by profxj** [comment]
`test_setups.py`

================== 21 passed, 22 warnings in 90.38s (0:01:30) ==================

File: pypeit/spectrographs/keck_kcwi.py
Comment (kbwestfall) [inline pypeit/spectrographs/keck_kcwi.py:41]: Do we need this line somewhere in keck_kcwi.py? 
```
self.meta['instrument'] = dict(ext=0, card='INSTRUME')
```

File: pypeit/spectrographs/spectrograph.py
Comment (kbwestfall) [inline pypeit/spectrographs/spectrograph.py:99]: Stick a period after "Header" on the first line.

**Review by kbwestfall — COMMENTED** [review]
There are still lots of spectrographs missing.  Can we provide header_name for all the setups in the dev-suite?

**Comment by profxj** [comment]
=============== 302 passed, 5091 warnings in 1421.52s (0:23:41) ================

File: pypeit/spectrographs/keck_kcwi.py
Comment (profxj) [reply pypeit/spectrographs/keck_kcwi.py:41]: good catch

File: pypeit/spectrographs/spectrograph.py
Comment (profxj) [reply pypeit/spectrographs/spectrograph.py:99]: added

**Comment by profxj** [comment]
================== 21 passed, 22 warnings in 86.50s (0:01:26) ==================

**Review by kbwestfall — APPROVED** [review]
Thanks for filling in more spectrographs!

**Review by jhennawi — APPROVED** [review]
Make sure you merge in develop before you merge since you cannot merge now.

**Pull request merged**
